Providers/Profile.swift

```diff
@@ -135,7 +135,7 @@ class BrowserProfileSyncDelegate: SyncDelegate {
 /**
  * A Profile manages access to the user's data.
  */
-protocol Profile: class {
+protocol Profile: class, FirefoxAccountDelegate {
```

Comment:
This coupling is not great.  I can think of two alternatives:

* broadcast that the Account state has changed to the App at large; there's some mechanism for this in iOS (notifications?).  I feel like we don't use this (resource leaks?), but I'd like to understand why we don't.
* have the things that advance the account (UI bits, unfortunately) do the flushing, much like you did before (but without driving the advancement -- just by checking timestamps, updated flags, etc).